base/parse.jl

```diff
@@ -339,4 +340,7 @@ tryparse_internal(::Type{T}, s::AbstractString, startpos::Int, endpos::Int, rais
     tryparse_internal(T, s, startpos, endpos, 10, raise)
 
 parse(::Type{T}, s::AbstractString) where T<:Union{Real,Complex} =
-    tryparse_internal(T, s, firstindex(s), lastindex(s), true)
+    tryparse_internal(T, s, firstindex(s), lastindex(s), true)::T
```

Comment:
Shouldn't this have a `convert(T, ...)` call?

Reply:
See my comment below - it fails to pass the tests (this is what I have used initially in my first commit)